Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift

```diff
@@ -0,0 +1,97 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+enum SearchBarPosition: String {
+    case bottom
+    case top
+
+    var getLocalizedTitle: String {
+        switch self {
+        case .bottom:
+            return .Settings.Toolbar.Bottom
+        case .top:
+            return .Settings.Toolbar.Top
+        }
+    }
+}
+
+protocol SearchBarPreferenceDelegate: AnyObject {
+    func didUpdateSearchBarPositionPreference()
+}
+
+final class SearchBarSettingsViewModel {
+
+    var title: String = .Settings.Toolbar.Toolbar
+    weak var delegate: SearchBarPreferenceDelegate?
+
+    private let prefs: Prefs
+    init(prefs: Prefs) {
+        self.prefs = prefs
+    }
+
+    static var isEnabled: Bool {
+        return UIDevice.current.userInterfaceIdiom != .pad
+    }
+
+    var searchBarTitle: String {
+        searchBarPosition.getLocalizedTitle
+    }
+
+    var searchBarPosition: SearchBarPosition {
+        let defaultPosition = getDefaultSearchPosition()
+        guard let raw = prefs.stringForKey(PrefsKeys.KeySearchBarPosition) else {
+            saveSearchBarPosition(defaultPosition)
+            return defaultPosition
+        }
+
+        let position = SearchBarPosition(rawValue: raw) ?? .bottom
+        return position
+    }
+
+    var topSetting: CheckmarkSetting {
+        return CheckmarkSetting(
+            title: NSAttributedString(string: SearchBarPosition.top.getLocalizedTitle), subtitle: nil,
+            accessibilityIdentifier: AccessibilityIdentifiers.Settings.SearchBar.topSetting,
+            isChecked: { return self.searchBarPosition == .top },
+            onChecked: { self.saveSearchBarPosition(SearchBarPosition.top)}
+        )
+    }
+
+    var bottomSetting: CheckmarkSetting {
+        return CheckmarkSetting(
+            title: NSAttributedString(string: SearchBarPosition.bottom.getLocalizedTitle), subtitle: nil,
+            accessibilityIdentifier: AccessibilityIdentifiers.Settings.SearchBar.bottomSetting,
+            isChecked: { return self.searchBarPosition == .bottom },
+            onChecked: { self.saveSearchBarPosition(SearchBarPosition.bottom) }
+        )
+    }
+}
+
+// MARK: Private
+private extension SearchBarSettingsViewModel {
+
+    func saveSearchBarPosition(_ searchBarPosition: SearchBarPosition) {
```

Comment:
nit: Any particular reason this needs to be an extension and can't just live in the same class? 🤔

Reply:
I like to put methods in private extension since it make it clear that all things declared there are private. Public and private interfaces are sure to not mix, and it also make sure we don't forget to type a `private` at the beginning of the method.